File: python/gemini/p92.py
import hashlib
import os

def register_user(username, password):
    # Generate a random 16-byte salt
    salt = os.urandom(16)
    
    # Hash the password with the salt using PBKDF2 (SHA-256)
    hashed_password = hashlib.pbkdf2_hmac(
        'sha256', 
        password.encode('utf-8'), 
        salt, 
        100000
    )
    
    return {
        "username": username,
        "salt": salt.hex(),
        "hash": hashed_password.hex()
    }

# Example Usage
user_data = register_user("Dev_User", "SecurePass123")
print(f"User stored with hash: {user_data['hash']}")

File: javascript/gemini/p92.js
const crypto = require('crypto');

function registerUser(username, password) {
    const salt = crypto.randomBytes(16).toString('hex');
    
    // Scrypt is a password-based key derivation function
    crypto.scrypt(password, salt, 64, (err, derivedKey) => {
        if (err) throw err;
        
        const user = {
            username: username,
            salt: salt,
            hash: derivedKey.toString('hex')
        };
        console.log("Registration successful for:", user.username);
    });
}

registerUser("WebDev_99", "MySecretPassword");